packages/design-system/src/components/OcResourceIcon/OcResourceIcon.vue

```diff
@@ -122,4 +129,38 @@ export default defineComponent({
     height: 70%;
   }
 }
+.oc-resource-icon-status-badge {
+  position: absolute;
+  bottom: -3px;
+  right: -2px;
+  width: 12px;
+  height: 12px;
+  padding: var(--oc-space-xsmall);
+  border-radius: 30px;
+  transition: background-color 200ms ease-in-out;
+  background: var(--oc-color-background-default);
+
+  &-inner {
+    position: absolute;
+    top: 2px;
+    left: 2px;
+    width: 8px;
+    height: 8px;
+    padding: var(--oc-space-xsmall);
+    line-height: var(--oc-space-small);
+    border-radius: 30px;
+  }
+  .oc-icon {
+    margin-top: -2px;
+    margin-left: -1.5px;
+    svg {
+      fill: var(--oc-color-background-default) !important;
+    }
+  }
+  .oc-spinner {
+    margin-left: -2px;
+    margin-top: -2px;
+    color: var(--oc-color-background-default) !important;
+  }
```

Comment:
Any chance to abstract this code? Right now, this component is making assumptions for what's being rendered into the slot. IMO that should be handled either in a generic way (just target spans maybe?), or in the component filling the slot.

Reply:
hm yeah, but if I move it in the component where the OcResourceIcon is used we have duplicated styling. I get your point but I would leave it here.